src/System.Web.HttpUtility/src/System/Web/HttpUtility.cs

```diff
@@ -142,52 +143,70 @@ public static NameValueCollection ParseQueryString(string query, Encoding encodi
             return result;
         }
 
-        public static string HtmlDecode(string s) => HttpEncoder.HtmlDecode(s);
+        [return: NotNullIfNotNull("s")]
+        public static string? HtmlDecode(string? s) => HttpEncoder.HtmlDecode(s);
 
-        public static void HtmlDecode(string s, TextWriter output) => HttpEncoder.HtmlDecode(s, output);
+        public static void HtmlDecode(string? s, TextWriter output) => HttpEncoder.HtmlDecode(s, output);
 
-        public static string HtmlEncode(string s) => HttpEncoder.HtmlEncode(s);
+        [return: NotNullIfNotNull("s")]
+        public static string? HtmlEncode(string? s) => HttpEncoder.HtmlEncode(s);
 
-        public static string HtmlEncode(object value) =>
-            value == null ? null : HtmlEncode(Convert.ToString(value, CultureInfo.CurrentCulture));
+        [return: NotNullIfNotNull("value")]
+        public static string? HtmlEncode(object? value) =>
+            value == null ? null : HtmlEncode(Convert.ToString(value, CultureInfo.CurrentCulture) ?? string.Empty);
 
-        public static void HtmlEncode(string s, TextWriter output) => HttpEncoder.HtmlEncode(s, output);
+        public static void HtmlEncode(string? s, TextWriter output) => HttpEncoder.HtmlEncode(s, output);
 
-        public static string HtmlAttributeEncode(string s) => HttpEncoder.HtmlAttributeEncode(s);
+        [return: NotNullIfNotNull("s")]
+        public static string? HtmlAttributeEncode(string? s) => HttpEncoder.HtmlAttributeEncode(s);
 
-        public static void HtmlAttributeEncode(string s, TextWriter output) => HttpEncoder.HtmlAttributeEncode(s, output);
+        public static void HtmlAttributeEncode(string? s, TextWriter output) => HttpEncoder.HtmlAttributeEncode(s, output);
 
-        public static string UrlEncode(string str) => str == null ? null : UrlEncode(str, Encoding.UTF8);
+        [return: NotNullIfNotNull("str")]
+        public static string? UrlEncode(string? str) => UrlEncode(str, Encoding.UTF8);
```

Comment:
I removed the `str == null ? null :` here and a few places below where it was unnecessary: the function it's calling to does the exact same check, and we shouldn't be optimizing for the null input case with additional code.